specification/dartLangSpec.tex

```diff
@@ -3154,11 +3158,44 @@ \subsubsection{Generative Constructors}
 <initializerExpression> ::= <conditionalExpression> | <cascade>
 \end{grammar}
 
+\LMHash{}%
+If an \synt{initializerExpression} is a \synt{functionExpression},
+a compile-time error occurs.
+
+\rationale{%
+The reason for having this restriction is that
+it may be necessary to scan the code
+to the end of a potentially long function body in order to determine
+whether that function body is the body of the constructor,
+or it is the body of a function literal
+which is an \synt{initializerExpression}.
+This may impede the readability of the code for human beings,
+and it may cause performance issues for parsers.
+Also, such a function literal may capture variables from
+the formal parameter initializer scope
+(\ref{generativeConstructors}),
+which may be error prone because such a variable has
+the same name as an instance field,
+but it is a distinct variable and may have a different value.%
+}
+
+\commentary{%
+Assuming that this restriction did not exist, consider the following example:
+`\code{C():\,\,\THIS.y\,\,=\,(x)\,\{\ldots\}}'.
+If that snippet of code is followed by `\code{;}' or `\code{,}'
+then `\code{(x)\,\{\ldots\}}' is a function literal,
+otherwise it is the expression \code{(x)} followed by the constructor body.
+If needed, the restriction may be avoided
+by adding a parenthesis around the function literal,
+`\code{C():\,\,\THIS.y\,\,=\,((x)\,\{\ldots\})}'.%
+}
+
 \LMHash{}%
```

Comment:
I assume the changes below are reflows only.

Reply:
That is true, I generally bring a unit of text (like a subsection) into a "80 chars at most" format when it is edited, trying to maintain the unity of meaning (such that grep will still be able to produce useful results).